js/classes/entities/foreground/weather_system.js

```diff
@@ -1,52 +1,49 @@
+const weatherType = {
+    HOT: "hot",
+    WARM: "warm",
+    CLOUDS: "clouds",
+    RAIN: "rain",
+    SNOW: "snow"
+};
+
 /**
  * WeatherSystem manages what weather to spawn into a zone.
  * 
  * @author Caleb Krauter
  */
 class WeatherSystem {
 
-    /**
-     * Intensity is used for both rain/snow and clouds simultaneously.
-     * 
-     * @param {String} type warm, hot, cloudy, rain, snow.
-     * @param {int} intensity a value between 0 and 5.
-     * @param {String} time day or night.
-     */
-    constructor(type, intensity, time) {
-        this.type = type;
-        this.intensity = intensity;
-        this.time = time;
-        this.clouds = false;
-
-        WeatherSystem.weatherType = {
-            HOT: "hot",
-            WARM: "warm",
-            CLOUDS: "clouds",
-            RAIN: "rain",
-            SNOW: "snow"
-        };
-        if (this.type === WeatherSystem.weatherType.RAIN) {
-            this.makeClouds();
-            this.makePrecipitation();
+    static setWeather(type, intensity, time, startingYOfClouds = 19) {
+        WeatherSystem.type = type;
+        WeatherSystem.intensity = intensity;
+        WeatherSystem.time = time;
+        WeatherSystem.clouds = false;
+        WeatherSystem.isRain = WeatherSystem.type === weatherType.RAIN;
+        WeatherSystem.isSnow = WeatherSystem.type === weatherType.SNOW;
+        WeatherSystem.isClouds = WeatherSystem.type === weatherType.CLOUDS;
+        WeatherSystem.startingYOfClouds = startingYOfClouds;
```

Comment:
You shouldn't be using static fields - you don't need anything to persist. Instead, just pass these values as parameters to the different methods that need them.